### PR TITLE
Site Title: Add border block support 

### PR DESCRIPTION
**Pull request by akasunil**
## What?

Add border block support to the `Site Title` block.

Part of https://github.com/WordPress/gutenberg/issues/43247

## Why?
`Site Title` block is missing border support.

## How?

Adds the border block support in block.json

## Testing Instructions

- Go to Global Styles setting ( under appearance > editor > styles > edit styles > blocks )
- Make sure that `Site Title` block's border is configurable via Global Styles
- Verify that Global Styles are applied correctly in the editor and frontend
- Edit template/page, Add `Site Title` block and Apply the border styles
- Verify that block borders take precedence over global styles
- Verify that block borders display correctly in both the editor and frontend

## Screenshots or screencast 
In backend:
Screenshot at Jul 16 22-30-29
In frontend: 
Screenshot at Jul 16 22-30-50

### CODE REVIEW

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for working on this @akasunil 👍 

This tests well for me. 

https://github.com/user-attachments/assets/0d3fe2f7-264b-4c9a-a7b1-3a9d77aaa0dc




The only thing I think we might need is the adoption of the `box-sizing: border-box` for more consistent size/styling of the block. It seems like adoption of this is pretty inconsistent across blocks at the moment, so I'd like to get some second opinions. 

cc @andrewserong

**Comment by andrewserong** [comment]
> The only thing I think we might need is the adoption of the box-sizing: border-box for more consistent size/styling of the block. It seems like adoption of this is pretty inconsistent across blocks at the moment, so I'd like to get some second opinions.

Thanks for the ping! Adding the border-box rule sounds consistent with other similar blocks, e.g. the post title block where it was introduced when padding support was added back in: https://github.com/WordPress/gutenberg/pull/43457.

The site title padding support was added way back in https://github.com/WordPress/gutenberg/pull/31125 so has been around for a long time.

Are there any concerns that adding box-sizing at this stage could affect sites out in the wild? If so, one option could be to guard the rule so it only applies if a border is present. That said, it _feels_ more like a fix to me to add in the rule.

I'll just ping @carolinan for visibility on this one as Carolina introduced padding for this block originally and might have thoughts on the matter.

**Comment by aaronrobertshaw** [comment]
> That said, it feels more like a fix to me to add in the rule.

Agreed. 

> Are there any concerns that adding box-sizing at this stage could affect sites out in the wild?

I think it would have to be a rather small edge case where the site title block is given an explicit width or inherits one from its parent only to have padding pushing its effective width wider.

File: packages/block-library/src/site-title/style.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
+:root :where(.wp-block-site-title) {
```
Comment (aaronrobertshaw) [inline packages/block-library/src/site-title/style.scss]:
```suggestion
.wp-block-site-title {
```
The `:root :where()` selector results in `0-1-0` specificity the same as `.wp-block-site-title` just using more characters and data. Perhaps we should just simplify that?

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for the quick update making the border controls here optional 🚀 

I've given this another test and all is working as expected. I've left one small nit as an inline comment. Once we tweak that, this is looking ready to me. Nice work 👍

File: packages/block-library/src/site-title/style.scss (hunk 1 above)
Comment (akasunil) [reply packages/block-library/src/site-title/style.scss]: Sure

Comment (aaronrobertshaw) [reply packages/block-library/src/site-title/style.scss]: Sorry @akasunil, I meant only for the `box-sizing` rule. The change in https://github.com/WordPress/gutenberg/pull/63631/commits/f880412530c690e0e0f4877d6d59a9564cce1dfa bumps the specificity of the `a` rule to `0-1-1`.

**Review by aaronrobertshaw — APPROVED** [review]
This is testing well for the adoption of border support 👍 

✅ Global border styles apply correctly
✅ Block instance styles override global styles
✅ Box sizing takes into account border when parent enforces width

I restored the specificity for the inner link rule as the recent change in selector accidentally bumped it from `0-1-0` to `0-1-1`. 


https://github.com/user-attachments/assets/9adc784d-9dc3-4dd4-ac7b-c05ae0ffc365

File: packages/block-library/src/site-title/style.scss (hunk 1 above)
Comment (aaronrobertshaw) [reply packages/block-library/src/site-title/style.scss]: To save another feedback cycle and review, I've tweaked the selector in https://github.com/WordPress/gutenberg/pull/63631/commits/95c4f7baba0b5506cea5f8bc5d5703a676eaaa23

**Pull request merged**